packages/apollo-gateway/src/index.ts

```diff
@@ -73,6 +85,38 @@ function isManagedConfig(
   return !isRemoteConfig(config) && !isLocalConfig(config);
 }
 
+type DidResolveQueryPlanCallback = ({
+  queryPlan,
+  serviceMap,
+  operationContext,
+}: {
+  queryPlan: QueryPlan;
+  serviceMap: ServiceMap;
+  operationContext: OperationContext;
+}) => void;
+
+type DidFailCompositionCallback = ({
```

Comment:
I think we also wanted to have the composition config ID in there? Perhaps nullably, but some ID to correspond to (in this case, it can be captured and routed from the managed config)

Reply:
I think we'll definitely want the metadata from the config that's downloaded, which contains an ID and the cloud-computed schema hash. That will be a useful resource in figuring out what failed and what _should_ have happened :)

Reply:
A good suggestion! Resolved in 868c0b499bdc9c7802d35533d5ef37ea984d25be